integrations/mistral/tests/test_mistral_chat_generator.py

```diff
@@ -187,7 +198,12 @@ def test_check_abnormal_completions(self, caplog):
         ]
 
         for m in messages:
-            component._check_finish_reason(m)
+            try:
+                # Haystack >= 2.9.0
+                component._check_finish_reason(m.meta)
+            except AttributeError:
+                # Haystack < 2.9.0
+                component._check_finish_reason(m)
```

Comment:
this is ugly, but we can get rid of it after releasing Haystack 2.9.0
(I'll create an issue to track that)